core/integration/module_private_deps_test.go

```diff
@@ -0,0 +1,79 @@
+package core
+
+import (
+	"context"
+	"testing"
+
+	"github.com/dagger/testctx"
+	"github.com/stretchr/testify/require"
+)
+
+type PrivateDepsSuite struct{}
+
+func TestPrivateDeps(t *testing.T) {
+	testctx.New(t, Middleware()...).RunTests(PrivateDepsSuite{})
+}
+
+func (PrivateDepsSuite) TestPrivateDeps(ctx context.Context, t *testctx.T) {
+	t.Run("golang", func(ctx context.Context, t *testctx.T) {
+		privateDepCode := `package main
+
+import (
+        "context"
+        "dagger/foo/internal/dagger"
+
+        "github.com/rajatjindal/bkpapi/backend/pkg/api"
```

Comment:
Hi @jedevc, I am currently using one of my private repo for simulating a private dep, but I think we can add a private go module in our [dagger-tests-modules](https://github.com/dagger/dagger-test-modules) repo and use the `private forked` version of that in this test.